splink/connected_components.py

```diff
@@ -268,10 +284,57 @@ def _cc_assess_exit_condition(representatives_name):
     return sql
 
 
+def _cc_create_unique_id_cols(linker, match_probability_threshold):
+
+    """
+    Takes the output of linker.predict() and either creates unique IDs for
+    our linked dataframes, if we are performing a link job, or pulls out
+    the unique ID columns if deduping.
+    """
+
+    df_predict = linker.predict()
+
+    # Code assumes a unique ID column is provided.
+    if linker._settings_dict["link_type"] == "dedupe_only":
+        sql = f"""
+            select unique_id_l, unique_id_r
+            from {df_predict.physical_name}
+            where match_probability >= {match_probability_threshold}
+
+            UNION
+
+            select
+            unique_id as unique_id_l,
+            unique_id as unique_id_r
+            from __splink__df_concat_with_tf
+        """
+
+    else:
```

Comment:
If we use `linker._settings_obj._unique_id_input_columns` carefully we can get rid of the else